src/Account.ts

```diff
@@ -0,0 +1,322 @@
+import { concat, defer, first, switchMap, type ObservableInput } from 'rxjs'
+import { ABI } from './abi/index.js'
+import type { Centrifuge } from './Centrifuge.js'
+import { NULL_ADDRESS } from './constants.js'
+import { Entity } from './Entity.js'
+import type { HexString } from './types/index.js'
+import { repeatOnEvents } from './utils/rx.js'
+import { doSignMessage, doTransaction, signPermit } from './utils/transaction.js'
+
+const tUSD = '0x8503b4452Bf6238cC76CdbEE223b46d7196b1c93'
+
+export class Account extends Entity {
+  constructor(
+    _root: Centrifuge,
+    public accountId: HexString,
+    public chainId: number
+  ) {
+    super(_root, ['account', accountId, chainId])
+  }
+
+  balances() {
+    return this._query(['balances'], () => {
+      return defer(async () => {
+        const client = this._root.getClient(this.chainId)!
+        const balance = (await client.readContract({
+          address: tUSD,
+          abi: ABI.Currency,
+          functionName: 'balanceOf',
+          args: [this.accountId],
+        })) as bigint
+        return balance
+      }).pipe(
+        repeatOnEvents(
+          this._root,
+          {
+            address: tUSD,
+            abi: ABI.Currency,
+            eventName: 'Transfer',
+            filter: (events) => {
+              console.log('Transfer Event')
+              return events.some((event) => {
+                return event.args.from === this.accountId || event.args.to === this.accountId
+              })
+            },
+          },
+          this.chainId
+        )
+      )
+    })
+  }
+
+  transfer(to: HexString, amount: bigint) {
```

Comment:
seeing a lot of versions of the transfer function still, do you want to keep these or should we evaluate and choose the best?